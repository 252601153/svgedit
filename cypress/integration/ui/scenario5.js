import {
  visitAndApproveStorage
} from '../../support/ui-test-helper.js';

const testSnapshot = () => {
  cy.get('#svgcontent').cleanSnapshot();
};

describe('use all parts of svg-edit', function () {
  before(() => {
    visitAndApproveStorage();
  });

  it('check tool_source_set', function () {
    cy.get('#tool_source').click({ force: true });
    cy.get('#svg_source_textarea')
      .type('{selectall}', { force: true })
      .type(`<svg width="640" height="480" xmlns="http://www.w3.org/2000/svg" xmlns:svg="http://www.w3.org/2000/svg">
      <g class="layer">
       <title>Layer 1</title>
       </g>
     </svg>`, { force: true, parseSpecialCharSequences: false });
    cy.get('#tool_source_save').click({ force: true });
    testSnapshot();
  });
  it('check tool_line', function () {
    cy.get('#tool_line')
      .click({ force: true });
    cy.get('#svgcontent')
      .trigger('mousemove', 200, 200, { force: true })
      .trigger('mousedown', 200, 200, { force: true })
      .trigger('mousemove', 250, 250, { force: true })
      .trigger('mouseup', { force: true });
    cy.get('#svgcontent').toMatchSnapshot();
  });
  it('check tool_line_change_class', function () {
    cy.get('#svg_1').click({ force: true });
    cy.get('#elem_class').shadow().find('elix-input').eq(0).shadow().find('#inner').eq(0)
      .type('svg_1_class{enter}', { force: true });
    cy.get('#svg_1')
      .should('satisfy', ($el) => {
        const classList = Array.from($el[0].classList);
        return classList.includes('svg_1_class');
      });
  });
  it('check tool_line_change_id', function () {
    cy.get('#svg_1').click({ force: true }).click({ force: true });
    cy.get('#elem_id').shadow().find('elix-input').eq(0).shadow().find('#inner').eq(0)
      .type('_id{enter}', { force: true });
    cy.get('#svg_1_id')
      .should('satisfy', ($el) => {
        const classList = Array.from($el[0].classList);
        return classList.includes('svg_1_class');
      });
  });
  it('check tool_line_change_rotation', function () {
    cy.get('#svg_1_id').click({ force: true });
    for(let n = 0; n < 5; n ++){
      cy.get('#angle').shadow().find('elix-number-spin-box').eq(0).shadow().find('#upButton').eq(0)
        .click({ force: true });
    }
    cy.get('#svgcontent').toMatchSnapshot();
  });
  it('check tool_line_change_blur', function () {
    cy.get('#svg_1_id').click({ force: true });
    for(let n = 0; n < 10; n ++){
      cy.get('#blur').shadow().find('elix-number-spin-box').eq(0).shadow().find('#upButton').eq(0)
        .click({ force: true });
    }
    cy.get('#svgcontent').toMatchSnapshot();
  });
  it('check tool_line_change_opacity', function () {
    cy.get('#svg_1_id').click({ force: true });
    for(let n = 0; n < 10; n ++){
      cy.get('#opacity').shadow().find('elix-number-spin-box').eq(0).shadow().find('#downButton').eq(0)
        .click({ force: true });
    }
    cy.get('#svgcontent').toMatchSnapshot();
  });
  it('check tool_line_delete', function () {
    cy.get('#svg_1_id').click({ force: true });
    cy.get('#tool_delete').click({ force: true });
    cy.get('#svgcontent').toMatchSnapshot();
  });
  it('check tool_line_clone', function () {
    cy.get('#tool_line')
      .click({ force: true });
    cy.get('#svgcontent')
      .trigger('mousemove', 200, 200, { force: true })
      .trigger('mousedown', 200, 200, { force: true })
      .trigger('mousemove', 250, 250, { force: true })
      .trigger('mouseup', { force: true });
    cy.get('#svg_2').click({ force: true });
    cy.get('#tool_clone').click({ force: true });
    cy.get('#svgcontent').toMatchSnapshot();
  });
  it('check tool_fhrect_change_x_y_coordinate', function () {
    cy.get('#svg_2').click({ force: true });
    for(let n = 0; n < 25; n ++){
      cy.get('#line_x1').shadow().find('elix-number-spin-box').eq(0).shadow().find('#upButton').eq(0)
        .click({ force: true });
    }
    for(let n = 0; n < 25; n ++){
      cy.get('#line_y1').shadow().find('elix-number-spin-box').eq(0).shadow().find('#downButton').eq(0)
        .click({ force: true });
    }
    for(let n = 0; n < 25; n ++){
      cy.get('#line_x2').shadow().find('elix-number-spin-box').eq(0).shadow().find('#upButton').eq(0)
        .click({ force: true });
    }
    for(let n = 0; n < 25; n ++){
      cy.get('#line_y2').shadow().find('elix-number-spin-box').eq(0).shadow().find('#downButton').eq(0)
        .click({ force: true });
    }
    cy.get('#svgcontent').toMatchSnapshot();
  });
});
