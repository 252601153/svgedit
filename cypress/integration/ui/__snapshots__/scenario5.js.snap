exports[`use all parts of svg-edit > check tool_source_set #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer"><title>Layer 1</title></g>
</svg>
`;

exports[`use all parts of svg-edit > check tool_line #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer" style="pointer-events:all">
    <title style="pointer-events:inherit">Layer 1</title>
    <line
      fill="none"
      stroke="#000000"
      stroke-width="5"
      opacity="0.5"
      style="pointer-events:none"
      x1="200"
      y1="200"
      x2="450"
      y2="450"
      id="svg_1"
    >
      <animate
        attributeName="opacity"
        begin="indefinite"
        dur="0.2"
        fill="freeze"
        to="1"
      ></animate>
    </line>
  </g>
</svg>
`;

exports[`use all parts of svg-edit > check tool_line_change_rotation #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer" style="pointer-events:all">
    <title style="pointer-events:inherit">Layer 1</title>
    <line
      fill="none"
      stroke="#000000"
      stroke-width="5"
      x1="200"
      y1="200"
      x2="450"
      y2="450"
      id="svg_1_id"
      fill-opacity="1"
      stroke-opacity="1"
      class="svg_1_class"
      transform="rotate(25 325 325)"
    ></line>
  </g>
</svg>
`;

exports[`use all parts of svg-edit > check tool_line_change_blur #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer" style="pointer-events:all">
    <title style="pointer-events:inherit">Layer 1</title>
    <line
      fill="none"
      stroke="#000000"
      stroke-width="5"
      x1="200"
      y1="200"
      x2="450"
      y2="450"
      id="svg_1_id"
      fill-opacity="1"
      stroke-opacity="1"
      class="svg_1_class"
      transform="rotate(25 325 325)"
      filter="url(#svg_1_id_blur)"
    ></line>
  </g>
  <defs>
    <filter id="svg_1_id_blur" x="-50%" y="-50%" width="200%" height="200%">
      <feGaussianBlur in="SourceGraphic" stdDeviation="5"></feGaussianBlur>
    </filter>
  </defs>
</svg>
`;

exports[`use all parts of svg-edit > check tool_line_change_opacity #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer" style="pointer-events:all">
    <title style="pointer-events:inherit">Layer 1</title>
    <line
      fill="none"
      stroke="#000000"
      stroke-width="5"
      x1="200"
      y1="200"
      x2="450"
      y2="450"
      id="svg_1_id"
      fill-opacity="1"
      stroke-opacity="1"
      class="svg_1_class"
      filter="url(#svg_1_id_blur)"
      opacity="0.5"
      transform="rotate(25 325 325)"
    ></line>
  </g>
  <defs>
    <filter id="svg_1_id_blur" x="-50%" y="-50%" width="200%" height="200%">
      <feGaussianBlur in="SourceGraphic" stdDeviation="5"></feGaussianBlur>
    </filter>
  </defs>
</svg>
`;

exports[`use all parts of svg-edit > check tool_line_delete #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer" style="pointer-events:all">
    <title style="pointer-events:inherit">Layer 1</title>
  </g>
  <defs>
    <filter id="svg_1_id_blur" x="-50%" y="-50%" width="200%" height="200%">
      <feGaussianBlur in="SourceGraphic" stdDeviation="5"></feGaussianBlur>
    </filter>
  </defs>
</svg>
`;

exports[`use all parts of svg-edit > check tool_line_clone #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer" style="pointer-events:all">
    <title style="pointer-events:inherit">Layer 1</title>
    <line
      fill="none"
      stroke="#000000"
      stroke-width="5"
      opacity="0.25"
      style="pointer-events:none"
      x1="200"
      y1="200"
      x2="450"
      y2="450"
      id="svg_2"
      fill-opacity="1"
      stroke-opacity="1"
    >
      <animate
        attributeName="opacity"
        begin="indefinite"
        dur="0.2"
        fill="freeze"
        to="0.5"
      ></animate>
    </line>
    <line
      fill="none"
      stroke="#000000"
      stroke-width="5"
      opacity="0.25"
      style="pointer-events:none"
      x1="220"
      y1="220"
      x2="470"
      y2="470"
      fill-opacity="1"
      stroke-opacity="1"
      id="svg_3"
    >
      <animate
        attributeName="opacity"
        begin="indefinite"
        dur="0.2"
        fill="freeze"
        to="0.5"
        id="svg_4"
      ></animate>
    </line>
  </g>
  <defs>
    <filter id="svg_1_id_blur" x="-50%" y="-50%" width="200%" height="200%">
      <feGaussianBlur in="SourceGraphic" stdDeviation="5"></feGaussianBlur>
    </filter>
  </defs>
</svg>
`;

exports[`use all parts of svg-edit > check tool_line_bring_to_back #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer" style="pointer-events:all">
    <title style="pointer-events:inherit">Layer 1</title>
    <line
      fill="none"
      stroke="#000000"
      stroke-width="5"
      opacity="0.5"
      x1="200"
      y1="200"
      x2="450"
      y2="450"
      id="svg_2"
      fill-opacity="1"
      stroke-opacity="1"
    ></line>
    <line
      fill="none"
      stroke="#000000"
      stroke-width="5"
      opacity="0.25"
      style="pointer-events:none"
      x1="220"
      y1="220"
      x2="470"
      y2="470"
      fill-opacity="1"
      stroke-opacity="1"
      id="svg_3"
    >
      <animate
        attributeName="opacity"
        begin="indefinite"
        dur="0.2"
        fill="freeze"
        to="0.5"
        id="svg_4"
      ></animate>
    </line>
  </g>
  <defs>
    <filter id="svg_1_id_blur" x="-50%" y="-50%" width="200%" height="200%">
      <feGaussianBlur in="SourceGraphic" stdDeviation="5"></feGaussianBlur>
    </filter>
  </defs>
</svg>
`;

exports[`use all parts of svg-edit > check tool_line_bring_to_front #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer" style="pointer-events:all">
    <title style="pointer-events:inherit">Layer 1</title>
    <line
      fill="none"
      stroke="#000000"
      stroke-width="5"
      opacity="0.25"
      style="pointer-events:none"
      x1="220"
      y1="220"
      x2="470"
      y2="470"
      fill-opacity="1"
      stroke-opacity="1"
      id="svg_3"
    >
      <animate
        attributeName="opacity"
        begin="indefinite"
        dur="0.2"
        fill="freeze"
        to="0.5"
        id="svg_4"
      ></animate>
    </line>
    <line
      fill="none"
      stroke="#000000"
      stroke-width="5"
      opacity="0.5"
      x1="200"
      y1="200"
      x2="450"
      y2="450"
      id="svg_2"
      fill-opacity="1"
      stroke-opacity="1"
    ></line>
  </g>
  <defs>
    <filter id="svg_1_id_blur" x="-50%" y="-50%" width="200%" height="200%">
      <feGaussianBlur in="SourceGraphic" stdDeviation="5"></feGaussianBlur>
    </filter>
  </defs>
</svg>
`;

exports[`use all parts of svg-edit > check tool_line_change_x_y_coordinate #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer" style="pointer-events:all">
    <title style="pointer-events:inherit">Layer 1</title>
    <line
      fill="none"
      stroke="#000000"
      stroke-width="5"
      opacity="0.25"
      style="pointer-events:none"
      x1="220"
      y1="220"
      x2="470"
      y2="470"
      fill-opacity="1"
      stroke-opacity="1"
      id="svg_3"
    >
      <animate
        attributeName="opacity"
        begin="indefinite"
        dur="0.2"
        fill="freeze"
        to="0.5"
        id="svg_4"
      ></animate>
    </line>
    <line
      fill="none"
      stroke="#000000"
      stroke-width="5"
      opacity="0.5"
      x1="225"
      y1="175"
      x2="475"
      y2="425"
      id="svg_2"
      fill-opacity="1"
      stroke-opacity="1"
    ></line>
  </g>
  <defs>
    <filter id="svg_1_id_blur" x="-50%" y="-50%" width="200%" height="200%">
      <feGaussianBlur in="SourceGraphic" stdDeviation="5"></feGaussianBlur>
    </filter>
  </defs>
</svg>
`;

exports[`use all parts of svg-edit > check tool_line_change_stroke_width #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer" style="pointer-events:all">
    <title style="pointer-events:inherit">Layer 1</title>
    <line
      fill="none"
      stroke="#000000"
      stroke-width="5"
      opacity="0.25"
      style="pointer-events:none"
      x1="220"
      y1="220"
      x2="470"
      y2="470"
      fill-opacity="1"
      stroke-opacity="1"
      id="svg_3"
    >
      <animate
        attributeName="opacity"
        begin="indefinite"
        dur="0.2"
        fill="freeze"
        to="0.5"
        id="svg_4"
      ></animate>
    </line>
    <line
      fill="none"
      stroke="#000000"
      stroke-width="15"
      opacity="0.5"
      x1="225"
      y1="175"
      x2="475"
      y2="425"
      id="svg_2"
      fill-opacity="1"
      stroke-opacity="1"
    ></line>
  </g>
  <defs>
    <filter id="svg_1_id_blur" x="-50%" y="-50%" width="200%" height="200%">
      <feGaussianBlur in="SourceGraphic" stdDeviation="5"></feGaussianBlur>
    </filter>
  </defs>
</svg>
`;

exports[`use all parts of svg-edit > check tool_line_change_stoke_color #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer" style="pointer-events:all">
    <title style="pointer-events:inherit">Layer 1</title>
    <line
      fill="none"
      stroke="#bf5f00"
      stroke-width="5"
      opacity="0.25"
      x1="220"
      y1="220"
      x2="470"
      y2="470"
      fill-opacity="1"
      stroke-opacity="1"
      id="svg_3"
    >
      <animate
        attributeName="opacity"
        begin="indefinite"
        dur="0.2"
        fill="freeze"
        to="0.5"
        id="svg_4"
      ></animate>
    </line>
    <line
      fill="none"
      stroke="#000000"
      stroke-width="15"
      opacity="0.5"
      x1="225"
      y1="175"
      x2="475"
      y2="425"
      id="svg_2"
      fill-opacity="1"
      stroke-opacity="1"
    ></line>
  </g>
  <defs>
    <filter id="svg_1_id_blur" x="-50%" y="-50%" width="200%" height="200%">
      <feGaussianBlur in="SourceGraphic" stdDeviation="5"></feGaussianBlur>
    </filter>
  </defs>
</svg>
`;

exports[`use all parts of svg-edit > check tool_line_align_to_page #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer" style="pointer-events:all">
    <title style="pointer-events:inherit">Layer 1</title>
    <line
      fill="none"
      stroke="#bf5f00"
      stroke-width="5"
      opacity="0.25"
      x1="220"
      y1="220"
      x2="470"
      y2="470"
      fill-opacity="1"
      stroke-opacity="1"
      id="svg_3"
    >
      <animate
        attributeName="opacity"
        begin="indefinite"
        dur="0.2"
        fill="freeze"
        to="0.5"
        id="svg_4"
      ></animate>
    </line>
    <line
      fill="none"
      stroke="#000000"
      stroke-width="15"
      opacity="0.5"
      x1="225"
      y1="175"
      x2="475"
      y2="425"
      id="svg_2"
      fill-opacity="1"
      stroke-opacity="1"
    ></line>
  </g>
  <defs>
    <filter id="svg_1_id_blur" x="-50%" y="-50%" width="200%" height="200%">
      <feGaussianBlur in="SourceGraphic" stdDeviation="5"></feGaussianBlur>
    </filter>
  </defs>
</svg>
`;
